### DISCUSSION TITLE
[Unknown Mod][Combat] Option to choose the troop that will play when several troops of the same speed are controlled by the hero during combat [Removed]

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#2786] [Ideas]
Add an option to choose the troop that will play when several troops of the same speed are controlled by the hero during combat:

"Battle: allows to choose the troop that will play when several troops of the same speed are controlled by the hero"

When a hero-controlled troop gets the hand and there are several hero-controlled troops that have the same speed, the player can choose which troop will play first among the troops that have not yet played in the current turn.
The controlled troop can be a troop from his army or an opposing troop controlled by the Hypnotize spell.

The troop that gets the hand is indicated by a yellow border flashing as usual.

The other troops controlled with the same speed are indicated by a different colored border; the color of the border is orange in the example for the Battle Dwarves and the opposing Pikemens (controlled by the Hypnotize spell).

When the player moves the mouse cursor over the sprite of one of these selectable troops, the question mark "?" cursor is replaced by another cursor to indicate to the player that this troop is selectable.
To select one of the other troops to play, the player simply left-clicks on the desired troop.
(For the opposing troop controlled by the Hypnotize spell, the hexagon is divided into several areas so that it is always possible to attack it depending on the position of the cursor on the hexagon.)

As long as the newly selected troop has not performed an action, the player can decide to select another one of these troops.
When the selected troop has performed its action, everything goes on as usual for the selection of the next troop to have the hand.
If the player doesn't select another troop, the troop that is active by default acts.

![Hypnotize fh2 all selected choice with orange 02](https://user-images.githubusercontent.com/43583994/107713512-b1147180-6ccb-11eb-9d25-7b086ae7b298.png)

**Comment by Branikolog** [comment]
That's a really strange idea, deeply rebuilding battle mechanics. Nothing similar was never implemented in any HoMM series....

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> That's a really strange idea, deeply rebuilding battle mechanics.

This comes from having played the game a lot.
Sometimes during the combat, being able to play a ranged troop instead of a hand-to-hand troop (and vice versa) would be really useful.

The term "deeply" is perhaps a bit exaggerated here. :D

> Nothing similar was never implemented in any HoMM series....

It's time to do it. :)

**Reply by Branikolog** [reply]
> Sometimes during the combat, being able to play a ranged troop instead of a hand-to-hand troop (and vice versa) would be really useful.

Yes, the advantages are obvious... But HoMM is a turn based game... Creatures gain their turn one by one. That's why we have to place them according their speed in hero garrison. I mean, if we want them to act sooner. That's the point. 
With this improvement we won't be able to cover shooters in most cases... It would be complete mess in understanding, who is going to attack first...
Anyway... So drastic changeover would turn battle into something different...
Probably, you've played the game for so long and became bored a bit, and would enjoy this nodification... But I'm personally not so veteran player, to start playing completely different game.))

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > Sometimes during the combat, being able to play a ranged troop instead of a hand-to-hand troop (and vice versa) would be really useful.
> 
> Yes, the advantages are obvious... But HoMM is a turn based game... Creatures gain their turn one by one. That's why we have to place them according their speed in hero garrison. I mean, if we want them to act sooner. That's the point.

This is part of a mod, so there will be changes.
And this is optional.

> With this improvement we won't be able to cover shooters in most cases...

It is quite the opposite.

> It would be complete mess in understanding, who is going to attack first...

Nothing complicated to understand.
The player chooses the troop that will play according to those who have the same speed.

> Anyway... So drastic changeover would turn battle into something different...
> Probably, you've played the game for so long and became bored a bit, and would enjoy this nodification...

This is purely speculative, and no I'm not bored by the game, I just want to improve it.

> But I'm personally not so veteran player, to start playing completely different game.))

So don't use this option.

**Reply by Branikolog** [reply]
@LeHerosInconnu , I don't want to offend you. 
It's just an opinion of your potential mod-user, so you could more precisely satisfy the needs of players and invent maximum comfortable mods.) 
For sure, other people could love this option.

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> @LeHerosInconnu , I don't want to offend you.

No worries. :)

> It's just an opinion of your potential mod-user, so you could more precisely satisfy the needs of players and invent maximum comfortable mods.)
> For sure, other people could love this option.

As fheroes2 allows to activate or not an option (and this is a strong point of fheroes2) and as this mod is intended to improve the gameplay according to my vision after having practiced a lot Heroes 2 and Heroes 3, but without imposing anything on the player (because everything is optional, which I consider important in a mod, because each player is different and will find interest or not to play with the proposed options), I don't see what the problem is.

If the player finds the option interesting, he keeps the option turned on.
If he doesn't like it, he turns it off.

Another thing to consider is that a mod should bring some innovation to the game and allow other approaches to be explored.
If everything remains the same compared to the original game, what is the point of proposing a mod?

**Reply by Branikolog** [reply]
> If everything remains the same compared to the original game, what is the point of proposing a mod?

That is a good statement! )
But I think, adding of new features should be delicate. I mean, if we add too much options, that might scare a player. That we can see in Era mod for H3. 
Also, adding new features is not a trivial task, so you should choose the best, the most valuable and the most enjoyable.

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > If everything remains the same compared to the original game, what is the point of proposing a mod?
> 
> That is a good statement! )
> But I think, adding of new features should be delicate. I mean, if we add too much options, that might scare a player. That we can see in Era mod for H3.
> Also, adding new features is not a trivial task, so you should choose the best, the most valuable and the most enjoyable.

Of course, a list of 1000 options one after the other is not necessarily encouraging.
But, it's more a problem of interface, how the options are presented, than anything else.

By grouping the options in sets, it becomes more pleasant to manage.
The options can also be grouped in a "mod".

The player can then look at the description of the mod and decide if he wants to try it or not.
He can then adapt the mod according to his wishes and disable the options that don't suit him.

The ability to customize the options is really a plus for players in my opinion.
And sometimes a simple little option can change a player's life and make all the difference between a good gaming experience and a not so good one.

**Comment by LeHerosInconnu** [comment]
Removed from the mod.
I close the discussion.
